lib/bamboo/api_error.ex

```diff
@@ -0,0 +1,34 @@
+defmodule Bamboo.ApiError do
+  @moduledoc """
+  Error used to represent a problem when sending emails through an external email service API.
+  """
+
+  defexception [:message]
+
+  def raise_api_error(message), do: raise(__MODULE__, message: message)
+  def raise_api_error(adapter, response, params, extra_message \\ "") when is_atom(adapter) do
+    message = """
+    There was a problem sending the email through the #{adapter_name(adapter)} API.
```

Comment:
@derekprior I though about this too, but for me it also seems wired to have a string, which is in some way related to what the module is, as the first argument of the function (`raise_api_error("Mailgun", ...)`)
What do you think of having a `@service_name` attribute on every adapter module and passing it to the function like so `raise_api_error(@service_name, ...)` ?